compiler/src/main/java/org/qbicc/graph/CallNoReturn.java

```diff
@@ -0,0 +1,84 @@
+package org.qbicc.graph;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.qbicc.type.FunctionType;
+import org.qbicc.type.definition.element.ExecutableElement;
+
+/**
+ *
+ */
+public final class CallNoReturn extends AbstractTerminator {
+    private final Node dependency;
+    private final BasicBlock terminatedBlock;
+    private final ValueHandle target;
+    private final List<Value> arguments;
+    private final FunctionType functionType;
+
+    CallNoReturn(Node callSite, ExecutableElement element, int line, int bci, final BlockEntry blockEntry, Node dependency, ValueHandle target, List<Value> arguments) {
+        super(callSite, element, line, bci);
+        this.dependency = dependency;
+        this.terminatedBlock = new BasicBlock(blockEntry, this);
+        this.target = target;
+        this.arguments = arguments;
+        functionType = (FunctionType) target.getValueType();
+    }
+
+    @Override
+    int calcHashCode() {
+        return Objects.hash(CallNoReturn.class, dependency, target, arguments);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other instanceof CallNoReturn && equals((CallNoReturn) other);
+    }
+
+    public boolean equals(CallNoReturn other) {
+        return this == other || other != null && dependency.equals(other.dependency) && target.equals(other.target) && arguments.equals(other.arguments);
```

Comment:
I think it's probably redundant.  If the dependency is the same then the terminated block must also be the same, since a terminator can only depend on its block's `BlockEntry` or one of its successors.